### PR TITLE
Implemented modifying table comments with `ALTER TABLE t MODIFY COMMENT 'value'`

### PR DESCRIPTION
**Pull request by Enmk**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implemented `ALTER TABLE x MODIFY COMMENT`

Detailed description / Documentation draft:
Users can now add\remove\modify comments on existing tables.
 - `ALTER TABLE x MODIFY COMMENT 'new comment'` - sets table comment to be a `new comment`, regardless if it was set before or not.
 - `ALTER TABLE x MODIFY COMMENT ''` - removes a table comment.

Comment change is reflected in both `system.tables` and `SHOW CREATE TABLE`.

### CODE REVIEW

File: tests/queries/0_stateless/2020_alter_table_modify_comment.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    get_table_comment_info
+
+    echo add a comment back
+    $CLICKHOUSE_CLIENT --query="ALTER TABLE comment_test_table MODIFY COMMENT 'another comment on a table';"
+    get_table_comment_info
+}
+
+test_table_comments "Null"
+test_table_comments "Memory"
+test_table_comments "MergeTree() ORDER BY k"
```
Comment (alesapin) [inline tests/queries/0_stateless/2020_alter_table_modify_comment.sh:48]: Please, add a test with ReplicatedMergeTree.

File: tests/queries/0_stateless/2020_alter_table_modify_comment.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    echo change a comment
+    $CLICKHOUSE_CLIENT --query="ALTER TABLE comment_test_table MODIFY COMMENT 'new comment on a table';"
+    get_table_comment_info
+
+    echo remove a comment
+    $CLICKHOUSE_CLIENT --query="ALTER TABLE comment_test_table MODIFY COMMENT '';"
+    get_table_comment_info
+
+    echo add a comment back
+    $CLICKHOUSE_CLIENT --query="ALTER TABLE comment_test_table MODIFY COMMENT 'another comment on a table';"
```
Comment (alesapin) [inline tests/queries/0_stateless/2020_alter_table_modify_comment.sh:42]: Add detach table/attach table and check one more time.

**Review by alesapin — CHANGES_REQUESTED** [review]

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
Integration tests not affected.

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-15458

**Comment by alexey-milovidov** [comment]
@alesapin, it's funny that even the test in this PR is wrong. It "cemented" the incorrect behavior.

**Comment by alexey-milovidov** [comment]
@Enmk, it does not work at all.

**Pull request merged**